samples/langchain_quick_start.ipynb

```diff
@@ -20,14 +22,1338 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from google.cloud import storage\n",
+    "import csv\n",
+    "from io import BytesIO\n",
+    "\n",
+    "\n",
+    "def read_csv_from_gcs(bucket_name, file_name):\n",
+    "    # Initialize a client\n",
+    "    storage_client = storage.Client()\n",
+    "\n",
+    "    # Get the bucket\n",
+    "    bucket = storage_client.get_bucket(bucket_name)\n",
+    "\n",
+    "    # Get the blob (file)\n",
+    "    blob = bucket.blob(file_name)\n",
+    "\n",
+    "    # Download the file\n",
+    "    content = blob.download_as_string()\n",
+    "\n",
+    "    # Read CSV content\n",
+    "    csv_data = content.decode(\"utf-8\")\n",
+    "    csv_reader = csv.reader(csv_data.splitlines())\n",
+    "\n",
+    "    # Convert CSV reader to list of rows\n",
+    "    rows = list(csv_reader)\n",
+    "\n",
+    "    return rows"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<google.api_core.operation.Operation at 0x7f188b238650>"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from google.cloud import spanner\n",
+    "\n",
+    "client = spanner.Client()\n",
+    "instance = client.instance(instance_id)\n",
+    "database = instance.database(database_id)\n",
+    "table_name = \"netflix_titles\"\n",
+    "\n",
+    "database.update_ddl(\n",
+    "    [\n",
+    "        \"\"\"CREATE TABLE IF NOT EXISTS {} (\n",
+    "    show_id STRING(MAX) NOT NULL,\n",
+    "    type STRING(MAX),\n",
+    "    title STRING(MAX),\n",
+    "    director STRING(MAX),\n",
+    "    casta STRING(MAX),\n",
+    "    country STRING(MAX),\n",
+    "    date_added STRING(MAX),\n",
+    "    release_year INT64,\n",
+    "    rating STRING(MAX),\n",
+    "    duration STRING(MAX),\n",
+    "    listed_in STRING(MAX),\n",
+    "    description STRING(MAX)\n",
+    ") PRIMARY KEY (show_id)\"\"\".format(\n",
+    "            table_name\n",
+    "        )\n",
+    "    ]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 63,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rows = read_csv_from_gcs(\n",
+    "    \"cloud-samples-data\", \"langchain/netflix_titles_compute_embeddings.csv\"\n",
+    ")\n",
+    "\n",
+    "batch_size = 500\n",
+    "\n",
+    "for i in range(1, len(rows), batch_size):\n",
+    "    with database.batch() as batch:\n",
+    "        batch_rows = rows[i : i + batch_size]\n",
+    "        batch.insert(\n",
+    "            table=table_name,\n",
+    "            columns=[\n",
+    "                \"show_id\",\n",
+    "                \"type\",\n",
+    "                \"title\",\n",
+    "                \"director\",\n",
+    "                \"casta\",\n",
```

Comment:
typo?

Reply:
cast is reserved keyword, updating it to `cast_list`